pkg/corerp/api/v20231001preview/container_conversion.go

```diff
@@ -151,6 +156,64 @@ func (src *ContainerResource) ConvertTo() (v1.DataModelInterface, error) {
 	return converted, nil
 }
 
+// toEnvDataModel: Converts from versioned datamodel to base datamodel
+func toEnvDataModel(e map[string]*EnvironmentVariable) (map[string]datamodel.EnvironmentVariable, error) {
+
+	m := map[string]datamodel.EnvironmentVariable{}
+
+	for key, val := range e {
+		if val == nil {
+			return nil, v1.NewClientErrInvalidRequest(fmt.Sprintf("Environment variable %s is nil", key))
+		}
+		if val.Value != nil && val.ValueFrom != nil {
+			return nil, v1.NewClientErrInvalidRequest(fmt.Sprintf("Environment variable %s has both value and secret value", key))
+		}
```

Comment:
Could you add some information about **ValueFrom** and **Value** here?

Reply:
Added comment as requested

Reply:
Is the **valueFrom** the one from here: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#use-configmap-defined-environment-variables-in-pod-commands?